cli/pcluster/config/validators.py

```diff
@@ -593,7 +593,36 @@ def url_validator(param_key, param_value, pcluster_config):
             bucket, key = match.group(1), match.group(2)
             boto3.client("s3").head_object(Bucket=bucket, Key=key)
         except ClientError:
-            warnings.append("The S3 object does not exist or you do not have access to it.")
+
+            # custom_chef_cookbook accepts S3 URIs. Check that the referenced bucket
+            # is in s3_read_resource or s3_read_write_resource.
+            if param_key == "custom_chef_cookbook":
```

Comment:
Let's remove this check. Your improvements are valid in general. 

The s3 uri can be used for pre and post install script parameters too.